src/emqx_broker.erl

```diff
@@ -436,9 +436,14 @@ init([Pool, Id]) ->
     true = gproc_pool:connect_worker(Pool, {Pool, Id}),
     {ok, #{pool => Pool, id => Id}}.
 
-handle_call({subscribe, Topic}, _From, State) ->
-    Ok = emqx_router:do_add_route(Topic),
-    {reply, Ok, State};
+handle_call({subscribe, Topic}, {FromPid, _Tag}, State) ->
```

Comment:
Sometimes we call broker:subscribe/2,3 from the test cases, and waiting for the PUBLISH messages being sent to its mailbox. I think they won't work after this change.